modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/server/persistence/RocksDbKeyValueStorage.java

```diff
@@ -1144,39 +1147,24 @@ private void addAllToBatch(
      * Adds modified entries to the watch event queue.
      */
     private void queueWatchEvent() {
-        if (updatedEntries.isEmpty()) {
```

Comment:
Because we can have a command to remove all keys that did not delete anything and we do not update records but at the same time update the revision and this code did not allow notification about updating the revision.